File: Day - 0/Weighted Mean/try.java
import java.util.Scanner;

public class WeightedMean {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int[] X = new int[n];
        int[] W = new int[n];

        for (int i = 0; i < n; i++) {
            X[i] = scanner.nextInt();
        }

        for (int i = 0; i < n; i++) {
            W[i] = scanner.nextInt();
        }

        weightedMean(X, W);
    }

    private static void weightedMean(int[] X, int[] W) {
        double wMean = 0;

        for (int i = 0; i < X.length; i++) {
            wMean += X[i] * W[i];
        }

        wMean /= sum(W);

        System.out.println(String.format("%.1f", wMean));
    }

    private static int sum(int[] arr) {
        int sum = 0;

        for (int num : arr) {
            sum += num;
        }

        return sum;
    }
}

File: Day - 0/Weighted Mean/try.c
#include <stdio.h>

double calculateWeightedMean(int X[], int W[], int n) {
    double wMean = 0;

    for (int i = 0; i < n; i++) {
        wMean += X[i] * W[i];
    }

    return wMean;
}

int calculateSum(int arr[], int n) {
    int sum = 0;

    for (int i = 0; i < n; i++) {
        sum += arr[i];
    }

    return sum;
}

int main() {
    int n;
    scanf("%d", &n);

    int X[n];
    int W[n];

    for (int i = 0; i < n; i++) {
        scanf("%d", &X[i]);
    }

    for (int i = 0; i < n; i++) {
        scanf("%d", &W[i]);
    }

    double wMean = calculateWeightedMean(X, W, n);
    int sumW = calculateSum(W, n);

    wMean /= sumW;

    printf("%.1f\n", wMean);

    return 0;
}
